FacesofLondon/app/src/main/java/com/novoda/wallpaper/droidcon/TimeOfDayCalculator.java

```diff
@@ -0,0 +1,34 @@
+package com.novoda.wallpaper.droidcon;
+
+import java.util.Calendar;
+import java.util.Locale;
+
+import static com.novoda.wallpaper.droidcon.TimeOfDay.*;
+
+class TimeOfDayCalculator {
+
+    TimeOfDay currentTimeOfDay() {
+        Calendar calendar = Calendar.getInstance();
+        int hourOfDay = calendar.get(Calendar.HOUR_OF_DAY);
+        return forHourOfDay(hourOfDay);
+    }
+
+    TimeOfDay forHourOfDay(int hourOfDay) {
+        mustBeValidHourOfDay(hourOfDay);
+        if (hourOfDay >= DAWN.hourOfDay() && hourOfDay < DAY.hourOfDay()) {
+            return DAWN;
+        } else if (hourOfDay >= DAY.hourOfDay() && hourOfDay < DUSK.hourOfDay()) {
+            return DAY;
+        } else if (hourOfDay >= DUSK.hourOfDay() && hourOfDay < NIGHT.hourOfDay()) {
+            return DUSK;
+        } else {
+            return NIGHT;
+        }
+    }
+
+    private static void mustBeValidHourOfDay(int hourOfDay) {
+        if (hourOfDay < 0 || hourOfDay > 24) {
```

Comment:
`Calendar.HOUR_OF_DAY` is the 24hour version of `Calendar.HOUR` 

See: https://developer.android.com/reference/java/util/Calendar.html#HOUR_OF_DAY